test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+
+    const eprosima::fastdds::dds::DomainParticipant * const_participant = participant;
+    const eprosima::fastdds::statistics::dds::DomainParticipant * const_statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_NE(const_statistics_participant, nullptr);
+
+    participant = nullptr;
+    statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    const_participant = nullptr;
+    const_statistics_participant = eprosima::fastdds::statistics::dds::DomainParticipant::narrow(const_participant);
+    EXPECT_EQ(const_statistics_participant, nullptr);
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_ERROR);
+#endif // FASTDDS_STATISTICS
+}
+
+/*
+ * This test checks both eprosima::fastdds::statistics::dds::DomainParticipant enable_statistics_datawriter() and
+ * disable_statistics_datawriter() methods.
+ * 1. If the FASTDDS_STATISTICS compilation flag is not set, check that they return RETCODE_UNSUPPORTED.
+ * 2. Create a eprosima::fastdds::dds::DomainParticipant and narrow to the children class.
+ * 3. Enable every single statistics datawriter successfully returning RETCODE_OK.
+ * 4. Check that the corresponding topic has been created and the type has been registered.
+ * 5. Enable and already enabled statistics DataWriter and check that it returns RETCODE_OK.
+ * 6. Call the method with an invalid topic name and check that returns RETCODE_BAD_PARAMETER.
+ * 7. Disable one statistics DataWriter and check that it is successful.
+ * 8. Enable the previous statistics DataWriter with an inconsistent QoS and check that it returns
+ * RETCODE_INCONSISTENT_POLICY.
+ * The case where the create_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
+ * upstream.
+ * 9. Try to delete the DomainParticipant and check that fails with RETCODE_PRECONDITION_NOT_MET because there are still
+ * statistics DataWriters enabled.
+ * 10. Try to disable an already disabled statistics DataWriter and check that returns RETCODE_ERROR.
+ * 11. Check that if an invalid topic name is provided to the disable_statistics_datawriter method, it returns
+ * RETCODE_BAD_PARAMETER.
+ * 12. Delete DDS entities.
+ * The case where the delete_datawriter fails returning RETCODE_ERROR is not checked because it only passes the error
+ * upstream.
+ */
+TEST(StatisticsDomainParticipantTests, EnableDisableStatisticsDataWriterTest)
+{
+eprosima::fastdds::dds::DataWriterQos inconsistent_qos = eprosima::fastdds::dds::DATAWRITER_QOS_DEFAULT;
+inconsistent_qos.reliability().kind = eprosima::fastdds::dds::BEST_EFFORT_RELIABILITY_QOS;
+inconsistent_qos.ownership().kind = eprosima::fastdds::dds::EXCLUSIVE_OWNERSHIP_QOS;
+
```

Comment:
Create standard participant here

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3